### PR TITLE
Stats: Add Edit/View links to Post Detail

### PR DESCRIPTION
**Pull request by timmyc**
This is a quick "try" branch to explore adding a few additional links to the Post Detail page.  @supernovia commented that it would be a nice addition to have the ability to view the post when looking at its stats.  She suggested we could use the iframe preview to do so - so that is what has been added here.

It also made sense to provide a quick link to edit the post as well.  I'm not certain the footer location is the best place for these links, but I was following the pattern we went with on the Summary page for the Download CSV button.

trout_bummin_ _wordpress_com

__To Test__
- Visit a site stats page
- Select a post from the "Post & Pages" component
- Interact with the buttons at the bottom of the screen

### CODE REVIEW

**Comment by folletto** [comment]
This is a really good idea. :)

My advice would be:

1. Move the "Edit" button inside the WebPreview modal (top-right corner).
2. I'd move the action button in the top bar.
3. There's a [query string](https://github.com/Automattic/wp-calypso/blob/47f1a198578f9827f73e42539baa4f112de1fac4/client/my-sites/themes/helpers.js#L20) to avoid the top admin bar to show. It's probably used by both the Editor and Themes (note that the two URLs for the external link and the content needs to be provided separately).

Mocks:

screen shot 2017-01-30 at 11 28 59

screen shot 2017-01-30 at 11 28 54



Also:

* It seems it doesn't work when the page clicked is a "Homepage" (id = 0) we might want to treat it differently so stats and link work?

**Comment by timmyc** [comment]
Well I had to get a bit creative to make `<HeaderCake />` work with a toolbar action on the right side.  There was an existing `<Button />` there being used as a spacer.  Not sure if my solution is the most elegant, but it works and doesn't break existing usage:

view-post

Small viewport:

edit_post_ _trout_bummin_ _wordpress_com

File: client/my-sites/stats/stats-post-detail/index.jsx [hunk 1]
```diff
@@ -86,6 +113,18 @@ class StatsPostDetail extends Component {
 				/>
 
 				<PostWeeks siteId={ this.props.siteId } postId={ this.props.postId } />
+				<WebPreview
```
Comment (youknowriad) [inline client/my-sites/stats/stats-post-detail/index.jsx]: I think all websites are not previewable, (Jetpack non https and also iframes can be explicitely blocked). We had issues with that when working on taxonomies previews. I think now there's a Component that take care of these issues `WithPreviewProps`. More background in #10205

File: client/components/header-cake/index.jsx [hunk 2]
```diff
@@ -47,6 +52,10 @@ HeaderCake.propTypes = {
 	onTitleClick: PropTypes.func,
 	backText: PropTypes.string,
 	backHref: PropTypes.string,
+	actionText: PropTypes.string,
+	actionHref: PropTypes.string,
+	actionIcon: PropTypes.string,
+	actionOnClick: PropTypes.func,
```
Comment (youknowriad) [inline client/components/header-cake/index.jsx]: Should we update the README with the new options here?

**Comment by folletto** [comment]
I gave it a try, design wise: 👍

File: client/my-sites/stats/stats-post-detail/index.jsx (hunk 1 above)
Comment (timmyc) [inline client/my-sites/stats/stats-post-detail/index.jsx]: Indeed, I should have disclosed that in the PR summary.  I had chatted with @supernovia about that when we discussed this feature in slack.  The exposure here is a bit _less_ than in the editor, as posts in stats have been published.

Granted there could still be issues using the web preview with private posts on JP sites, and the user not being logged in there... but it won't always fail like viewing a draft on a JP site does.

File: client/components/header-cake/index.jsx (hunk 2 above)
Comment (timmyc) [inline client/components/header-cake/index.jsx]: Absolutely!  I was waiting to make sure the solution was acceptable prior to updating the README ;)

File: client/my-sites/stats/stats-post-detail/index.jsx (hunk 1 above)
Comment (timmyc) [inline client/my-sites/stats/stats-post-detail/index.jsx]: The only issue I see with the `WithPreviewProps` approach is we lose the ability to have the Edit button in the WebPreview toolbar.  I might just rely upon a combination of `get( site, 'is_previewable' ) && ! isJetpack` and conditionally use WebPreview or an external link.

**Comment by timmyc** [comment]
In 3763ec3 the 'View' button is only shown when a site `is_previewable` and `! isJetpack`.  Documentation and dev docs example of new action props were also added.

**Comment by youknowriad** [comment]
LGTM 👍

**Pull request merged**